### PR TITLE
Feature/feature prep python3

### PR DESCRIPTION
**Pull request by nicoxx**

### CODE REVIEW

File: src/server/app.yaml [hunk 1]
```diff
@@ -17,3 +17,5 @@ resources:
 env_variables:
     GCLOUD_PROJECT: gator-01
 
+skip_files:
+    - ^tests$
```
Comment (m-mokhtari) [inline src/server/app.yaml]: As written in https://cloud.google.com/appengine/docs/python/config/appref#skip_files the default exclusions list must be kept to extends it. We must keep it to exclude .pyc, .pyo, files ending with ~ ...

File: tools/start_server.sh [hunk 2]
```diff
@@ -5,11 +5,11 @@
 
 # kill previous instance of flask server and free port if needed
 # cf. http://stackoverflow.com/questions/3510673/find-and-kill-a-process-in-one-line-using-bash-and-regex 
-kill $(ps aux | grep '[p]ython src/server/server/main.py' | awk '{print $2}')
+kill $(ps aux | grep '[p]ython src/server/main.py' | awk '{print $2}')
 fuser -k 8080/tcp
 # use appengine virtual env to ensure that flask server only uses python libs available, that is the ones 
 # defined in requirements.txt next to server python module.
-appengine_env/bin/python src/server/server/main.py &
+appengine_env/bin/python  src/server/main.py &
```
Comment (m-mokhtari) [inline tools/start_server.sh]: There is 2 spaces between "python" and "src".

File: src/learner/tests/test_learner.py [hunk 3]
```diff
@@ -6,6 +6,8 @@
 from learner.learner import UserDocumentsAccumulator, UserData, UserDoc, _similarity_by_row, _FixedSizeHeap
 
 # test implementation of L2 norm without re-using numpy code
+
```
Comment (m-mokhtari) [inline src/learner/tests/test_learner.py]: Why adding two blank lines ? am I missing something ? The above comment is not related to norm function ?

File: tools/start_gcloud_and_exit.sh [hunk 4]
```diff
@@ -3,9 +3,12 @@
...
-rm start_gcloud.log
+rm -f start_gcloud.log
 touch start_gcloud.log
 (tools/start_gcloud.sh 2>&1 | tee start_gcloud.log) &
 { tail -n +1 -f start_gcloud.log &} | sed -n '/Dev App Server is now running/q'
 # inspired by the trick explained here:
-# http://unix.stackexchange.com/questions/33018/have-bash-script-wait-for-status-message-before-continuing
\ No newline at end of file
+# http://unix.stackexchange.com/questions/33018/have-bash-script-wait-for-status-message-before-continuing
+
```
Comment (m-mokhtari) [inline tools/start_gcloud_and_exit.sh]: Why these blank lines ?

File: src/server/app.yaml (hunk 1 above)
Comment (nicoxx) [inline src/server/app.yaml]: well spot! Done

File: src/learner/tests/test_learner.py (hunk 3 above)
Comment (nicoxx) [inline src/learner/tests/test_learner.py]: autopep8 tool is a little overzealous...Fixed (comment should be inside function in python or it's a considered to be a module level comment I think

File: tools/start_gcloud_and_exit.sh (hunk 4 above)
Comment (nicoxx) [inline tools/start_gcloud_and_exit.sh]: fixed

**Pull request merged**